fixtures/css/sass_package_import.scss

```diff
@@ -0,0 +1,7 @@
+// There seems to be an issue with the sass-loader and prefixing
+// aliased directories with a tilde. The following line is how
+// this import should look and works for node_modules.
+// @import '~lib/test-pkg';
+
+// Importing without the tilde seems to work for webpack aliases
+@import 'lib/test-pkg';
```

Comment:
Can you walk me through this process for how this is resolved? I don't quite see the full picture yet:

1) Webpack sees that we're loading the `lib/test-pkg` module
2) But, it doesn't know yet (right?) if this is loading a JS file (silly example, but technically correct), a JSON file, a font file or a CSS file. So, how does it know to look at the `resolve` key from the css rule?

Reply:
When the sass-loader encounters `'lib/test-pkg'` it asks Webpack to perform the resolution of this import and it does so using the process outlined in the [Resolve](https://webpack.js.org/configuration/resolve/) configuration. Webpack will notice that lib is an alias to `./lib` and that `test-pkg` is a directory under it which contains a package.json file. To determine which file to import, Webpack evaluates each value in the `mainFields` resolve option in order to determine if it has been specified in the package.json file. If the `mainFields` resolve option is not specified, Webpack defaults to `browser`, `module`, and then `main` when the `target` property is set to `web`,`webworker`, or is unspecified. If none of these properties exist in the package.json file, then Webpack will revert to using extension based resolution. In this case since `'lib/text-pkg'` resolves to a directory, Webpack will look for a file in this directory using the names specified in `mainFiles` resolve option having an extension of one of the values specified in the `extensions` resolve option. The `mainFiles` resolve option defaults to `index` and the `extensions` resolve options defaults to `['.wasm', '.mjs', '.js', '.json']`. With the options below specified, Webpack will instead look for `index.scss`, `index.sass`, and then `index.css`.